python-package/lightgbm/basic.py

```diff
@@ -1515,7 +1515,8 @@ def _lazy_init(self, data, label=None, reference=None,
         # set feature names
         return self.set_feature_name(feature_name)
 
-    def __yield_row_from(self, seqs: List[Sequence], indices: Iterable[int]):
+    @staticmethod
```

Comment:
Because we don't use `self` in this method and the logic is quite general.

I considered moving it to `Sequence` interface as a static method when writing this PR. This would exposes the ability of indexing from list of `Sequence` to user.